src/devices/bus/mononcol/rom.h

```diff
@@ -0,0 +1,55 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli
+#ifndef MAME_BUS_MONONCOL_ROM_H
+#define MAME_BUS_MONONCOL_ROM_H
+
+#pragma once
+
+#include "slot.h"
+
+#include "machine/generic_spi_flash.h"
+
+// ======================> mononcol_rom_device
+
+class mononcol_rom_device : public device_t, public device_mononcol_cart_interface
+{
+protected:
+	// construction/destruction
+	mononcol_rom_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock);
+
+	// device-level overrides
+	virtual void device_start() override { }
```

Comment:
Please move the definition to the .cpp file, to make it clearer where the vtable will be instantiated (if the destructor is defaulted, the vtable is instantiated where the first out-of-line virtual member function is defined).

Reply:
at the risk of sounding like a broken record, this probably needs to be fixed in generic/rom.h, due to that being the goto example of how to use this, and where the code originated.

Reply:
Who exactly is putting it forward as the go-to example of how to write a cartridge slot?